### PR TITLE
Ban circular dependencies

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Comment by jakolehm** [comment]
I think this is way too big change for a patch release. Moving to next minor.

File: __mocks__/windowMock.ts [hunk 1]
```diff
@@ -19,14 +19,17 @@
...
+    writable: true,
+    value: jest.fn().mockImplementation(callback => callback()),
+  });
 
-export default {};
+  Object.defineProperty(window, "cancelIdleCallback", {
+    writable: true,
+    value: jest.fn(),
+  });
+}
```
Comment (aleksfront) [inline __mocks__/windowMock.ts]: @Nokel81 Thanks for solving this!

Comment (Nokel81) [reply __mocks__/windowMock.ts]: You are welcome! Hopefully the rest of this PR can be reviewed.

**Comment by Nokel81** [comment]
I starting to think that this is a bit big. Will split it up into several PRs.

**Pull request closed without merging**